synapse/storage/databases/main/events.py

```diff
@@ -2177,7 +2211,9 @@ def _remove_push_actions_for_event_id_txn(self, txn, room_id, event_id):
             (room_id, event_id),
         )
 
-    def _store_rejections_txn(self, txn, event_id, reason):
+    def _store_rejections_txn(
+        self, txn: LoggingTransaction, event_id: str, reason: Union[bool, str]
```

Comment:
I think `reason` will only ever be a `str`. This only has a single caller:

https://github.com/matrix-org/synapse/blob/320186319ac4f1d16f8f964d92db8921a4b1073e/synapse/storage/databases/main/events.py#L1482-L1486

And [`context.rejected` is](https://github.com/matrix-org/synapse/blob/e2e1d90a5e4030616a3de242cde26c0cfff4a6b5/synapse/events/snapshot.py#L37):

> A rejection reason if the event was rejected, else False

Reply:
If I am change this to `str`:
```
synapse/storage/databases/main/events.py:1491: error: Argument 3 to "_store_rejections_txn" of "PersistEventsStore" has incompatible type "Union[Literal[True], str]"; expected "str"  [arg-type]
```

One option ist to change
https://github.com/matrix-org/synapse/blob/aaaff98202fc1be4e3794291c4b670b8fc62cee5/synapse/storage/databases/main/events.py#L1485
to
```
self._store_rejections_txn(txn, event.event_id, str(context.rejected))
```

I think the reason is:
https://github.com/matrix-org/synapse/blob/aaaff98202fc1be4e3794291c4b670b8fc62cee5/synapse/events/snapshot.py#L109

Reply:
> And [`context.rejected` is](https://github.com/matrix-org/synapse/blob/e2e1d90a5e4030616a3de242cde26c0cfff4a6b5/synapse/events/snapshot.py#L37):
> 
> > A rejection reason if the event was rejected, else False

Assuming this is true, can you try `rejected: Union[Literal[False], str] = False`? on snapshot.py:109?

Reply:
This is a solution and works.